src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh

```diff
@@ -105,7 +106,7 @@ __vsc_prompt_cmd_original() {
 	for ((i = 0; i < ${#ADDR[@]}; i++)); do
 		builtin eval ${ADDR[i]}
 	done
-	IFS=''
+	IFS="$__vsc_original_ifs"
```

Comment:
I'm not entirely positive but I think you need to differentiate between the "IFS unset" case and the "IFS is empty string" case:

```
glasser@dsg-mbp 0 12:11:13 ~ $ SPLITME='foo bar baz'
glasser@dsg-mbp 0 12:11:15 ~ $ IFS=''
glasser@dsg-mbp 0 12:11:16 ~ $ for x in $SPLITME; do echo $x; done
foo bar baz
glasser@dsg-mbp 0 12:11:25 ~ $ unset IFS
glasser@dsg-mbp 0 12:11:29 ~ $ for x in $SPLITME; do echo $x; done
foo
bar
baz
glasser@dsg-mbp 0 12:11:30 ~ $ X="$IFS"
glasser@dsg-mbp 0 12:11:56 ~ $ IFS="$X"
glasser@dsg-mbp 0 12:12:05 ~ $ for x in $SPLITME; do echo $x; done
foo bar baz
```

As you can see, the two cases have different effects, but passing the value through another variable like you do in this file will convert from unset to empty.

Reply:
(You can check `${IFS+set}` to see if the variable is set: if it is set this evaluates to `set` and if it is not set it evaluates to empty.)